.circleci/config.yml

```diff
@@ -19,10 +19,10 @@ references:
     BASH_ENV: .circleci/bash_env.sh
     GO_VERSION: 1.20.1
   envoy-versions: &supported_envoy_versions
-    - &default_envoy_version "1.21.5"
-    - "1.22.5"
-    - "1.23.2"
-    - "1.24.0"
+    - &default_envoy_version "1.21.6"
```

Comment:
Doing this here might not be the correct place because of the migration to Github Actions?

Reply:
I chatted w/ @jmurret and am going to commit envoy version updates to https://github.com/hashicorp/consul/pull/16915

Reply:
Also chatted about a backport that incorrectly updated these versions. Temporarily reverting them here, but there's no real impact since this workflow will be removed shortly